src/engine/runtime.js

```diff
@@ -552,6 +570,7 @@ Runtime.prototype._step = function () {
     this.redrawRequested = false;
     var inactiveThreads = this.sequencer.stepThreads();
     this._updateGlows(inactiveThreads);
+    this._setThreadCount(this.threads.length);
```

Comment:
Note that there is a chance that a thread will both start and stop in the same frame. This usually happens when running a single block for example: the block executes and then gets scrubbed from `this.threads` immediately. For script glows, I addressed this by passing `inactiveThreads` in the line above to the update function; any threads that were just scrubbed from `this.threads` are still counted as "glowing" for this frame because they're reported in `inactiveThreads`.

In Scratch 2.0, it looks like the green flag/stop button glow does respond when running a single block. Which to me indicates that this line should maybe be:

`this._setThreadCount(this.threads.length + inactiveThreads.length);`

Then threads which were just scrubbed out of `this.threads` this frame will still be counted, until a full frame has passed.

P.S. maybe `inactiveThreads` here should be called `scrubbedThreads` or something.